File: store/src/main/java/com/kjj/store/client/ImageClient.java
package com.kjj.store.client;

import com.kjj.store.util.ClientUriTool;
import com.kjj.store.util.HttpTools;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;

@Service
@RequiredArgsConstructor
public class ImageClient {
    private final RestTemplate restTemplate;
    private final HttpTools httpTools;
    private final ClientUriTool clientUriTool;
    @Value("${my.client.image.host}") private String imageServerHost;

    public String uploadImage(MultipartFile file, String dir) throws IOException {
        return restTemplate.postForObject(
                clientUriTool.getRequestURI(imageServerHost, "/v1/image"),
                httpTools.getRequestEntityTypeFormData(file, dir),
                String.class);
    }

    public void updateImage(MultipartFile file, String dir) throws IOException {
        restTemplate.patchForObject(
                clientUriTool.getRequestURI(imageServerHost, "/v1/image"),
                httpTools.getRequestEntityTypeFormData(file, dir),
                Void.class);
    }
}

File: store/src/main/kotlin/com/kjj/store/client/ImageClient.kt
package com.kjj.store.client

import com.kjj.store.util.ClientUriTool
import com.kjj.store.util.HttpTool
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.web.client.RestTemplate
import org.springframework.web.client.patchForObject
import org.springframework.web.client.postForObject
import org.springframework.web.multipart.MultipartFile

@Service
class ImageClient(
    val restTemplate: RestTemplate,
    val httpTool: HttpTool,
    val clientUriTool: ClientUriTool
) {
    @Value("\${my.client.image.host}")
    lateinit var imageServerHost: String

    fun uploadImage(file: MultipartFile, dir: String): String? {
        return restTemplate.postForObject(
            clientUriTool.getRequestURI(imageServerHost, "/v1/image"),
            httpTool.getRequestEntityTypeFormData(file, dir),
            String::class
        )
    }

    fun updateImage(file: MultipartFile, dir: String) {
        restTemplate.patchForObject(
            clientUriTool.getRequestURI(imageServerHost, "/v1/image"),
            httpTool.getRequestEntityTypeFormData(file, dir),
            Unit::class.java
        )
    }
}